BigImageViewer/src/main/java/com/github/piasy/biv/view/BigImageView.java

```diff
@@ -342,17 +349,33 @@ public void showImage(Uri uri) {
     }
 
     public void showImage(final Uri thumbnail, final Uri uri) {
+        showImage(thumbnail, uri, false);
+    }
+
+    public void showImage(final Uri thumbnail, final Uri uri, final boolean delayImage) {
         mThumbnail = thumbnail;
         mUri = uri;
 
         clearThumbnailAndProgressIndicator();
-        mImageLoader.loadImage(hashCode(), uri, mInternalCallback);
+
+        mDelayImage = delayImage;
+        if (mDelayImage) {
+            mImageLoader.loadImage(hashCode(), thumbnail, mInternalCallback);
+        } else {
+            mImageLoader.loadImage(hashCode(), uri, mInternalCallback);
+        }
 
         if (mFailureImageView != null) {
             mFailureImageView.setVisibility(GONE);
         }
     }
 
+    public void triggerImageSwap() {
```

Comment:
Can we have a better name for this function? `triggerImageSwap` doesn't make much sense to me.

Reply:
I'm terrible with names in general, what can we call it?
showMainImage(), loadMainImage(), triggerMainImage(), loadDelayedImage()?

Reply:
Okay, I have a few suggestions:

* change the name of the third parameter of `showImage` to `delayMainImage`;
* change this function here to `loadMainImageNow`;
* change the name of `ImageCycleCallback` to `ImageShownCallback`, and it has two function: `onThumbnailShown` and `onMainImageShown`;

How do you think?

Reply:
consider it done

Reply:
Done